### PR TITLE
Substrate Companion for rust 1.54

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Review by ordian — APPROVED** [review]

**Comment by TriplEight** [comment]
@bkchr `-D warnings` generates new errors https://gitlab.parity.io/parity/polkadot/-/jobs/1109942

**Comment by TriplEight** [comment]
Pipelines are failing with new Rust versions.

**Review by thiolliere — APPROVED** [review]

**Comment by ordian** [comment]
bot cancel

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -15,7 +15,9 @@ stages:
   - publish
   - deploy
 
-image:                             paritytech/ci-linux:production
+# FIXME: revert me
+# image:                             paritytech/ci-linux:production
+image:                             paritytech/ci-linux:staging-1.54.0
```
Comment (ordian) [inline .gitlab-ci.yml:20]:
```suggestion
image:                             paritytech/ci-linux:production
```

**Comment by ordian** [comment]
bot merge

**Comment by TriplEight** [comment]
Now we should wait for https://gitlab.parity.io/parity/infrastructure/scripts/-/pipelines/159003 which will bring the stable 1.55

**Comment by ordian** [comment]
bot merge

**Comment by ordian** [comment]
https://gitlab.parity.io/parity/polkadot/-/jobs/1139228
> scripts/gitlab/check_extrinsics_ordering.sh: line 47: ./artifacts/polkadot: No such file or directory

**Comment by TriplEight** [comment]
> https://gitlab.parity.io/parity/polkadot/-/jobs/1139228
> 
> > scripts/gitlab/check_extrinsics_ordering.sh: line 47: ./artifacts/polkadot: No such file or directory

it [needs](https://gitlab.parity.io/parity/polkadot/-/blob/3807/.gitlab-ci.yml#L278-280) a polkadot bin from the previous job and it wasn't successful

**Comment by ordian** [comment]
> ./artifacts/polkadot: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by ./artifacts/polkadot)
./artifacts/polkadot: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ./artifacts/polkadot)

**Comment by TriplEight** [comment]
```
Using Docker executor with image node:15 ..
...
[1]-  Running                 $HEAD_BIN --chain="$RUNTIME-local" --tmp &
[2]+  Running                 $RELEASE_BIN --chain="$RUNTIME-local" --ws-port 9945 --tmp &
./artifacts/polkadot: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by ./artifacts/polkadot)
./artifacts/polkadot: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ./artifacts/polkadot)
```

**Comment by TriplEight** [comment]
We have `npm` installed in `paritytech/contracts-ci-linux:production`, it's `ubuntu:20.04` based let me replace it here.

**Comment by TriplEight** [comment]
now you should wait until https://github.com/paritytech/polkadot/pull/3807 gets merged

**Comment by ordian** [comment]
> Error: yargs supports a minimum Node.js version of 12. Read our version support policy: https://github.com/yargs/yargs#supported-nodejs-versions
    at Object.<anonymous> (/usr/local/lib/node_modules/@polkadot/metadata-cmp/node_modules/yargs/build/index.cjs:1:59111)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@polkadot/metadata-cmp/node_modules/yargs/index.cjs:5:30)
    at Module._compile (internal/modules/cjs/loader.js:778:30)

**Comment by TriplEight** [comment]
@ordian fixed in https://github.com/paritytech/polkadot/pull/3957

**Pull request merged**